src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -322,11 +322,11 @@ private void updateTransactionOnLoanClosure(StorageLoanDTO loan, InnReachTransac
 
       hold.setDueDateTime(null);
 
-      clearCentralPatronInfo(transaction.getHold());
-
       transaction.setState(FINAL_CHECKIN);
 
       notifier.reportFinalCheckIn(transaction);
+
+      clearCentralPatronInfo(hold);
```

Comment:
Can you please explain how this change is going to fix the issue with handling "finalcheckin" message on borrowing site?

Reply:
removed